### ISSUE TITLE
Proper handling of LVE process termination on Linux

### ISSUE THREAD

**Issue by d94pn** [modelica/Encryption-and-Licensing#3]
When calling 

```
mlle_start_executable
```
a child process is spawned but the PID is not considered and there is no corresponding mlle_stop_executable or similar that calls wait or waitpid for the exiting child process, as far as I can see. This will normally leave  zombie process where the chilld process (here the LVE) is wating for its parent to acknowledge its exiting. Is this merely neglected or have you any other means to avoid this?
I've not checked this on Windows so I don't know the status there.

**Comment by axelmartenssonmodelon** [comment]
This would be needed for the use case where the tool needs to unload a library without exiting. We have not considered this use case. Would you be interested in contributing with a PR?

**Comment by hubertus65** [comment]
My interpretation of this issue in the linked issue is that it can be closed as "won't fix" or similar. I'll leave it open until after the upcoming meeting.

**Comment by d94pn** [comment]
I suggest the following extension to handle this:

1. New member pid_child:

```
struct mlle_connections {
    int fd_to_child;    // Pipes. Used instead of streams
    int fd_from_child;  // that didn't work well with OpenSSL.
    SSL *ssl;
    int pid_child; // Process ID of child, needed for proper cleanup.
};
```

2. Set the pid_child in the implementation of mlle_start_executable.
3. Wait for the child process properly, calling waitpid() on the new pid_child member. I can think of 3 alternatives:  - Do it implicitly in mlle_connections_free.
  - Add an argument in mlle_connections_free setting the maximum waiting time in seconds.
  - Add a new function mlle_await_executable for doing the wait on the child process. Should also contain an argument with the maximum waiting time.

All 3 alternatives will require calling waitpid repeatedly with WNOHANG until either a value > 0 is returned or the wating time is consumed. For the first alternative a maximum waiting time should be hardcoded to something sensible, say 5 seconds.

I think the first alternative is good enough and is also least intrusive.
